flasc/floris_tools.py

```diff
@@ -11,302 +11,45 @@
-            raise DataError('Yaw should be defined in domain [0, 360) deg.')
-
-    # Split dataframe into subset dataframes for parallelization, if necessary
-    if num_workers > 1:
-        df_list = []
-
-        # See if we can simply split the problem up into a grid of conditions
-        num_jobs = num_workers * job_worker_ratio
-        n_unq = df["ws"].nunique() * df["wd"].nunique() * df["ti"].nunique()
-        if n_unq == df.shape[0]:
-            # Data is a grid of atmospheric conditions. Can divide and exploit
-            # the benefit of grid processing in floris v3.0.
-            Nconds_per_ti = df["ws"].nunique() * df["wd"].nunique()
-            Njobs_per_ti = int(np.floor(num_jobs / df["ti"].nunique()))
-            dN = int(np.ceil(Nconds_per_ti / Njobs_per_ti))
-
-            for ti in df["ti"].unique():
-                df_subset = df[df["ti"] == ti]
-                for ij in range(Njobs_per_ti):
-                    df_list.append(df_subset.iloc[(ij*dN):((ij+1)*dN)])
-                
-        else:
-            # If cannot be formatted to grid style, split blindly
-            dN = int(np.ceil(df.shape[0] / num_jobs))
-            for ij in range(num_jobs):
-                df_list.append(df.iloc[(ij*dN):((ij+1)*dN)])
-
-
-    # Calculate solutions
-    start_time = timerpc()
-    if num_workers <= 1:
-        print("Calculating floris solutions (non-parallelized)")
-        df_out = _run_fi_serial(
-            df_subset=df,
-            fi=fi,
-            include_unc=include_unc,
-            unc_pmfs=unc_pmfs,
-            unc_options=unc_options,
-            verbose=True
-        )
-    else:
-        print('Calculating with num_workers = %d and job_worker_ratio = %d'
-            % (num_workers, job_worker_ratio))
-        print('Each thread contains about %d FLORIS evaluations.' % dN)
-
-        # Define a tuple of arguments
-        multiargs = []
-        for df_mp in df_list:
-            df_mp = df_mp.reset_index(drop=True)
-            multiargs.append(
-                (df_mp, dcopy(fi), include_unc, unc_pmfs, unc_options, False)
-            )
-
-        if use_mpi:
-            # Use an MPI implementation, useful for HPC
-            from mpi4py.futures import MPIPoolExecutor as pool_executor
-        else:
-            # Use Pythons internal multiprocessing functionality
-            from multiprocessing import Pool as pool_executor
-
-        with pool_executor(num_workers) as pool:
-            df_list = pool.starmap(_run_fi_serial, multiargs)
-
-        df_out = pd.concat(df_list).reset_index(drop=True)
-        if 'index' in df_out.columns:
-            df_out = df_out.drop(columns='index')
-
-    t = timerpc() - start_time
-    print('Finished calculating the FLORIS solutions for the dataframe.')
-    print('Total wall time: %.3f s.' % t)
-    print('Mean wall time / function evaluation: %.3f s.' % (t/df.shape[0]))
-    return df_out
-
 
 def interpolate_floris_from_df_approx(
```

Comment:
Could you add a docstring here to for `interpolate_floris_from_df_approx()`? It doesn't need to be super detailed, but a quick explanation of the main functionality would be good.

Reply:
Added a docstring! Let me know if it suffices!

Reply:
That's a very impressive docstring :) Thanks!